### PR TITLE
Added a quick NIRISS parser test

### PR DESCRIPTION
**Pull request by rosteen**

### CODE REVIEW

File: jdaviz/configs/mosviz/tests/test_parsers.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import pytest
+import tempfile
+from zipfile import ZipFile
+import pathlib
+
+from astropy.utils.data import download_file
+
+from jdaviz.configs.mosviz.helper import MosViz
+
+@pytest.fixture
```
Comment (pllim) [inline jdaviz/configs/mosviz/tests/test_parsers.py]: This is unnecessary, use this:

https://github.com/spacetelescope/jdaviz/blob/519aaec9cc86b142e51d1b7f120ff70dd342124b/jdaviz/conftest.py#L28-L30

File: jdaviz/configs/mosviz/tests/test_parsers.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+def mosviz_helper():
+    return MosViz()
+
```
Comment (pllim) [inline jdaviz/configs/mosviz/tests/test_parsers.py:7]:
```suggestion


```

File: jdaviz/configs/mosviz/tests/test_parsers.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+@pytest.mark.remote_data
+def test_niriss_parser(mosviz_helper):
+    data_dir = tempfile.gettempdir()
```
Comment (pllim) [inline jdaviz/configs/mosviz/tests/test_parsers.py]: Use `tmpdir` or `tmp_path` fixture: https://docs.pytest.org/en/6.2.x/tmpdir.html

File: jdaviz/configs/mosviz/tests/test_parsers.py [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+    test_data = 'https://stsci.box.com/shared/static/9lkf5zha6zkf8ujnairy6krobbh038wt.zip'
```
Comment (pllim) [inline jdaviz/configs/mosviz/tests/test_parsers.py]: What is the size of this when uncompressed? Want to make sure we don't hit the CI quota.

If it is too big, we have to consider generating smaller set of fake data on the fly or using smaller download data set.

File: jdaviz/configs/mosviz/tests/test_parsers.py [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    with ZipFile(fn, 'r') as sample_data_zip:
+        sample_data_zip.extractall(data_dir)
+
+    level3_path = (pathlib.Path(data_dir) / 'NIRISS_for_parser_p0171')
+
+    data_dir = level3_path
+
+    mosviz_helper.load_niriss_data(data_dir)
+
+    assert len(mosviz_helper.app.data_collection) == 80
```
Comment (pllim) [inline jdaviz/configs/mosviz/tests/test_parsers.py]: 80 images? Seems excessive for a unit test.

File: jdaviz/configs/mosviz/tests/test_parsers.py (hunk 4 above)
Comment (rosteen) [reply jdaviz/configs/mosviz/tests/test_parsers.py]: Good call, that zip file contains a bunch of files not needed for this. I can upload a new zip file with only the files we need, which comes out to ~100MB. Is that big enough to cause problems?

Comment (astrofrog) [reply jdaviz/configs/mosviz/tests/test_parsers.py]: That should be ok

Comment (rosteen) [reply jdaviz/configs/mosviz/tests/test_parsers.py]: Alright, I'll leave the remaining files as-is for now then, I'll update the link shortly.

Comment (pllim) [reply jdaviz/configs/mosviz/tests/test_parsers.py]: Yes, 100 MB should be okay unless it decompressed into like 500 MB worth of files. 😉

File: jdaviz/configs/mosviz/tests/test_parsers.py [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+    with ZipFile(fn, 'r') as sample_data_zip:
+        sample_data_zip.extractall(tmpdir)
+
+    level3_path = (pathlib.Path(tmpdir) / 'NIRISS_for_parser_p0171')
+
+    data_dir = level3_path
+
+    mosviz_app.load_niriss_data(data_dir)
+
+    assert len(mosviz_app.app.data_collection) == 80
```
Comment (pllim) [inline jdaviz/configs/mosviz/tests/test_parsers.py:22]: Still don't know why we need 80 data entries to do this one test but if you said it is 100 MB, I'll let it go until CI complains...

**Review by pllim — APPROVED** [review]

File: jdaviz/configs/mosviz/tests/test_parsers.py (hunk 4 above)
Comment (rosteen) [reply jdaviz/configs/mosviz/tests/test_parsers.py]: Nah, 100 MB is the decompressed size.

File: jdaviz/configs/mosviz/tests/test_parsers.py (hunk 6 above)
Comment (rosteen) [reply jdaviz/configs/mosviz/tests/test_parsers.py:22]: Mainly it's the NIRISS dataset I have available, and I'm busy enough right now that I didn't want to go through the files to cut out the right HDUs that correspond to each other or make a dummy dataset with all the data types needed.

**Comment by rosteen** [comment]
Alright, I updated the link to use the smaller zip file I created (actually ended up at 54 MB unzipped). Still ends up with 80 entries in the data collection, sorry @pllim! I think this can be merged now that I did that @astrofrog .

**Comment by pllim** [comment]
It bothers me a bit to think that this could have been less than 10 MB if we cut down to like 2-5 datasets in the data collection. But just a bit. Your time is more precious at this point...

**Comment by rosteen** [comment]
I created a followup issue so your objection is not forgotten!

**Comment by astrofrog** [comment]
The download is not working for some reason.

**Comment by rosteen** [comment]
Looks like it wasn't just the NIRISS parser test that failed to download data, a Cubeviz and Imviz test with remote data failed as well.

**Comment by pllim** [comment]
STScI servers can be flaky. Try restart a few times.

**Comment by astrofrog** [comment]
I wonder if we should host these elsewhere, or it's going to be a pain for regular CI...

**Comment by pllim** [comment]
Last I heard, the "management" is discussing it...

**Comment by astrofrog** [comment]
Let's just merge in case somehow it works fine in the spacetelescope org

**Pull request merged**